common/djangoapps/course_modes/views.py

```diff
@@ -238,6 +238,14 @@ def get(self, request, course_id, error=None):  # lint-amnesty, pylint: disable=
         duration = get_user_course_duration(request.user, course)
         deadline = duration and get_user_course_expiration_date(request.user, course)
         context['audit_access_deadline'] = deadline
+        fbe_is_on = deadline and gated_content
+
+        # REV-2133 TODO Value Prop: remove waffle after testing is completed
+        # and happy path version is ready to be rolled out to all users.
+        if waffle.flag_is_active(request, 'course_modes.use_new_track_selection'):
+            if not error:  # Happy path does not handle errors.
```

Comment:
Could you say more about this comment? I'm thinking of happy path as the full experience for learners who are fbe and not enterprise (and if there are any errors that can happen in that flow, we'll handle them at the same standard as usual, right?)  Are there errors in this flow that we'll be intentionally ignoring? (not sure if this is inside another code block where they're possible) If yes could you say a little more about what they are?

Reply:
We'll revert to the current Track Selection page if there's some kind of enrollment error for now, even if they would be considered happy path. New page will actually have the error message be displayed at full screen and only the message itself in the page (as opposed to how it's now at the top of the page), UX will give a mockup hopefully.

So in the second iteration of this page we'll have non-happy path plus this error message version.

Reply:
that makes sense- thanks!